File: 14.Long_Common_Prefix/javascript/Long_Common_Prefix.js
var longestCommonPrefix = function (strs) {
  let res = ''
  for (let i = 0; i < strs.length; i++) {
    if (i === 0) {
      res = strs[i]
      continue
    }
    let length = Math.min(strs[i].length, res.length)
    let j = 0
    while (j < length && res[j] === strs[i][j]) j++
    res = res.slice(0, j)
  }
  return res
}
let strs = ['flower', 'flow', 'flight']
console.log(longestCommonPrefix(strs))

File: Easy/14.Long_Common_Prefix/java/Long_Common_Prefix.java
import java.util.HashMap;
import java.util.Map;

class Long_Common_Prefix {

    // other solution
    static public String longestCommonPrefix(String[] strs) {
        String res = "";
        for (int i = 0; i < strs.length; i++) {
            if (i == 0) {
                res = strs[0];
                continue;
            }
            int length = Math.min(res.length(), strs[i].length());
            int j = 0;
            while (j < length && strs[i].charAt(j) == res.charAt(j))
                j++;

            res = res.substring(0, j);

        }
        return res;
    }

    public static void main(String[] args) {
        String[] strs = { "flower", "flow", "flight" };
        System.out.println(longestCommonPrefix(strs));

    }
}